libyul/AsmParser.cpp

```diff
@@ -755,7 +755,13 @@ YulName Parser::expectAsmIdentifier()
 {
 	YulName name{currentLiteral()};
 	if (currentToken() == Token::Identifier && m_dialect.findBuiltin(name.str()))
```

Comment:
We could (and maybe should?) check for reserved identifiers here as well - what do you think?

Reply:
Not sure about that. It seems to me like a purely analysis-level check. It won't affect the correctness of the AST.

I was rather thinking about going the other way - moving the builtin check to analysis. It's not strictly necessary to parse things correctly. It safeguards against having ASTs where an identifier matches the name of a builtin so I guess it may be useful, but then that can be bypassed via AST import.